src/integer/vec_z/mul.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementation of the [`Mul`] trait for [`VecZ`] values.
+
+use super::{MatZ, VecZ};
+use crate::macros::arithmetics::{
+    arithmetic_trait_cross_types_borrowed_to_owned,
+    arithmetic_trait_cross_types_mixed_borrowed_owned,
+};
+use std::ops::Mul;
+
+impl Mul for &VecZ {
+    type Output = MatZ;
+
+    /// Implements the [`Mul`] trait for two [`VecZ`] values.
+    /// [`Mul`] is implemented for any combination of owned and borrowed [`MatZ`] and any kind of [`VecZ`].
+    ///
+    /// Parameters:
+    /// - `other`: specifies the value to multiply with `self`
+    ///
+    /// Returns the product of `self` and `other` as a [`MatZ`].
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer::VecZ;
+    /// use std::str::FromStr;
+    ///
+    /// let a = VecZ::from_str("[[2],[1],[1],[2]]").unwrap();
+    /// let b = VecZ::from_str("[[1,0,0,1]]").unwrap();
+    ///
+    /// let _ = &a * &b;
+    /// let _ = a.clone() * b.clone();
+    /// let _ = &a * b.clone();
+    /// let _ = a.clone() * &b;
+    /// ```
+    fn mul(self, other: Self) -> Self::Output {
+        (&self.matrix).mul(other)
```

Comment:
```suggestion
        // TODO: mul_safe
        (&self.matrix).mul(other)
```

Reply:
Why would I call mul_safe here if another function already provides exactly the behaviour I want to establish for this function.